TombEngine/Scripting/Internal/TEN/Effects/EffectsFunctions.cpp

```diff
@@ -133,10 +135,15 @@ namespace TEN::Scripting::Effects
 	static void EmitParticle(Vec3 pos, Vec3 velocity, int spriteIndex, TypeOrNil<int> gravity, TypeOrNil<float> rot, 
 							TypeOrNil<ScriptColor> startColor, TypeOrNil<ScriptColor> endColor, TypeOrNil<BlendMode> blendMode, 
 							TypeOrNil<int> startSize, TypeOrNil<int> endSize, TypeOrNil<float> lifetime, 
-							TypeOrNil<bool> damage, TypeOrNil<bool> poison)
+							TypeOrNil<bool> damage, TypeOrNil<bool> poison, TypeOrNil<GAME_OBJECT_ID> objectID, TypeOrNil<float> startRot)
 	{
-		if (!CheckIfSlotExists(ID_DEFAULT_SPRITES, "Particle spawn script function"))
-			return;
+		// Ensure objectID is valid 
+		GAME_OBJECT_ID effectiveObjectID = USE_IF_HAVE(GAME_OBJECT_ID, objectID, ID_DEFAULT_SPRITES); 
+		
+		// Validate the effective object ID 
+		if (!CheckIfSlotExists(effectiveObjectID, "Particle spawn script function"))
+		return;
```

Comment:
Is indentation missing here or is it incorrect GH formatting?

Reply:
This is how it looks at my end. Not sure.

![image](https://github.com/user-attachments/assets/92a56835-e58a-4b05-9b31-71108f6eda11)